src/main/java/ch/njol/skript/config/Node.java

```diff
@@ -150,6 +150,9 @@ static void handleNodeStackOverflow(StackOverflowError e, String line) {
 		
 		Skript.error("");
 		Skript.error("Updating your Java and/or using respective 64-bit versions for your operating system may also help and is always a good practice.");
+		Skript.error("If it is still not fixed, try changing the -Xss JVM option to a higher value (like -Xss2M) in your startup script.");
+		Skript.error("");
+		Skript.error("Using a different Java Virtual Machine (JVM) like OpenJ9 or GraalVM may also help; though be aware that not all plugins may support them.");
```

Comment:
Yes, they have different defaults and OpenJ9 has tail recursive optimization (and maybe a different Regex implementation/engine too), GraalVM has a new JIT compiler completely written in Java itself, and many more.

The message does only say "they may help", and that is true. If it is not fixed after trying different JVMs - then user perhaps can be sure that it is not JVMs fault, combining with the above recommendation. After that point it will be most likely that user had created a script specifically for exploding the stack and/or Skript's different unoptimized regexes take in place - or worse, an actual infinite repeat happening.

They are different in many areas - OpenJ9 uses same codebase for all Java versions for an example (so it takes advantage of VM enchancements in later releases), unlike HotSpot/OpenJDK which maintains different versions (8u, 11, 15, etc.) and backports important changes.

While these may not necessarily affect this problem - it will be worth to mention there in my opinion.

---------

After testing, I found out that with the same stack size, in OpenJ9 it does not happen on same input, though increasing the line, it will happen on OpenJ9 too. However the interesting part is, OpenJ9 gives the reason (the start of the call to the recursive function), but HotSpot does not. Not sure if it reserves stack space for the start of the call to the recursive function or it is because it eliminated tail recursive calls, making room for the Skript methods to show up in stack traces.

Found out another thing when testing too - ironically it gives this warning when on OpenJ9 (64-bit version for sure, it is false positive):
```
[04:37:35] [Server thread/WARN]: [Skript] You are currently using 32-bit Java. This may result in a StackOverflowError when loading aliases.
[04:37:35] [Server thread/WARN]: [Skript] Please update to 64-bit Java to remove this warning.
```

Will open a new issue and a different PR for that.